### PR TITLE
export xcoms parameters

### PR DESCRIPTION
**Pull request by benmackenzie-exos**
added code to extract metrics from the 'value' column in the xcom table and report as guage metrics. 

configurable in config.yaml as a collection of task_id, key pairs.

### CODE REVIEW

File: airflow_prometheus_exporter/__init__.py [hunk 1]
```diff
@@ -5,3 +5,17 @@
...
+
+dir = os.path.dirname(__file__)
+filename = os.path.join(dir, '../config.yaml')
+
+
+xcom_config = {}
+with open(filename) as file:
+    # The FullLoader parameter handles the conversion from YAML
+    # scalar values to Python the dictionary format
+    xcom_config = yaml.load(file, Loader=yaml.FullLoader)
```
Comment (abhishekray07) [inline airflow_prometheus_exporter/__init__.py:21]: I don't think this should be in `__init__.py`. Could we create a separate file for this?

File: airflow_prometheus_exporter/prometheus_exporter.py [hunk 2]
```diff
@@ -11,6 +11,12 @@
 from prometheus_client import generate_latest, REGISTRY
 from prometheus_client.core import GaugeMetricFamily
 from sqlalchemy import and_, func
+from airflow.configuration import conf
+from airflow_prometheus_exporter import xcom_config
+
+import pickle
+import json
+from airflow.utils.log.logging_mixin import LoggingMixin
```
Comment (abhishekray07) [inline airflow_prometheus_exporter/prometheus_exporter.py:19]: we try to follow the pep8 import rules to maintain the code as more readable (https://www.python.org/dev/peps/pep-0008/#imports) as well as maintaining alphabetical ordering

File: airflow_prometheus_exporter/prometheus_exporter.py [hunk 3]
```diff
@@ -149,6 +155,44 @@ def get_task_failure_counts():
             TaskFail.task_id,
         )
 
+def get_xcom_params(task_id):
+    #get latest task_id, join to xcom table
```
Comment (abhishekray07) [inline airflow_prometheus_exporter/prometheus_exporter.py:159]: Could we add docstring for the function?

File: airflow_prometheus_exporter/prometheus_exporter.py [hunk 4]
```diff
@@ -149,6 +155,44 @@ def get_task_failure_counts():
...
+    else:
+        try:
+            return json.loads(value.decode('UTF-8'))
+        except ValueError:
+            log = LoggingMixin().log
+            log.error("Could not deserialize the XCOM value from JSON. "
+                      "If you are using pickles instead of JSON "
+                      "for XCOM, then you need to enable pickle "
+                      "support for XCOM in your airflow config.")
+            raise
```
Comment (abhishekray07) [inline airflow_prometheus_exporter/prometheus_exporter.py:194]: What would happen if this raises an exception? Would this cause the exporter to fail?

**Review by abhishekray07 — CHANGES_REQUESTED** [review]
Could we also update the README to include instructions for how to use this?

File: airflow_prometheus_exporter/prometheus_exporter.py (hunk 4 above)
Comment (benmackenzie-exos) [reply airflow_prometheus_exporter/prometheus_exporter.py:194]: yes...good point...what if i just return an empty dict?

File: airflow_prometheus_exporter/prometheus_exporter.py (hunk 2 above)
Comment (benmackenzie-exos) [reply airflow_prometheus_exporter/prometheus_exporter.py:19]: ok

File: airflow_prometheus_exporter/prometheus_exporter.py (hunk 3 above)
Comment (benmackenzie-exos) [reply airflow_prometheus_exporter/prometheus_exporter.py:159]: yes

File: airflow_prometheus_exporter/__init__.py (hunk 1 above)
Comment (benmackenzie-exos) [reply airflow_prometheus_exporter/__init__.py:21]: ok

**Review by benmackenzie-exos — COMMENTED** [review]
i've made changes (i hope!)

**Comment by benmackenzie-exos** [comment]
actually...i created the pull request from the wrong repo.

**Pull request closed without merging**